source/game_sa/CarCtrl.cpp

```diff
@@ -379,9 +386,51 @@ void CCarCtrl::GenerateOneRandomCar() {
 
 // 0x4341C0
 void CCarCtrl::GenerateRandomCars() {
-    ZoneScoped;
+    if (CCutsceneMgr::ms_running) {
+        CountDownToCarsAtStart = 2;
+        return;
+    }
+
+    // check if a gang war is happening
+    if (CGangWars::DontCreateCivilians()) {
+        return;
+    }
+
+    // check if player is not in any interior
+    if (CGame::currArea != eAreaCodes::AREA_CODE_NORMAL_WORLD) {
+        return;
+    }
+
+    if (CGameLogic::LaRiotsActiveHere() && TimeNextMadDriverChaseCreated > 480.0f) {
+        TimeNextMadDriverChaseCreated = CGeneral::GetRandomNumberInRange(240.0f, 480.0f);
+    }
 
-    plugin::Call<0x4341C0>();
+    TimeNextMadDriverChaseCreated -= (CTimer::GetTimeStep() * 0.02f);
+
+    if (!(NumRandomCars >= 45)) {
+        int8_t _countDownToCarsAtStart = CountDownToCarsAtStart;
+
+        if (!_countDownToCarsAtStart) {
+            GenerateOneRandomCar();
+            GenerateOneRandomCar();
+        } else {
+            _countDownToCarsAtStart--;
+            CountDownToCarsAtStart = _countDownToCarsAtStart;
+
+            for (int i = 0; i <= 100; i++) {
+                GenerateOneRandomCar();
+            }
+
+            CTheCarGenerators::GenerateEvenIfPlayerIsCloseCounter = 20;
+        }
+    }
```

Comment:
No need for the extra variable here
```suggestion
    if (NumRandomCars < 45) {
        if (CountDownToCarsAtStart == 0) {
            GenerateOneRandomCar();
            GenerateOneRandomCar();
        } else {
            CountDownToCarsAtStart--;
            for (auto i = 100; i --> 0;) {
                GenerateOneRandomCar();
            }
            CTheCarGenerators::GenerateEvenIfPlayerIsCloseCounter = 20;
        }
    }
```